### PR TITLE
ENH: Add NPasteImageFilter

### PR DESCRIPTION
**Pull request by blowekamp**
This filter extends the PasteImageFilter with support to paste or fill
a constant into the defined region and support for the source image to
be a lower dimension than the input or destination image.

### CODE REVIEW

File: include/itkNPasteImageFilter.h [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+ * The SetDestinationIndex() method
+ * prescribes where in the destination input to start pasting data from the
+ * source input.  The SetSourceRegion method prescribes the section of
+ * the second image to paste into the first. When a constant pixel value is set, the SourceRegion describes the size
+ * of the region filled. If the output requested
+ * region does not include the SourceRegion after it has been
+ * repositioned to DestinationIndex, then the output will just be
+ * a copy of the input.
+ *
+ * This filter support running "InPlace" to efficiently reuses the destination image buffer for the output, removing the
```
Comment (zivy) [inline include/itkNPasteImageFilter.h]: support -> supports

File: include/itkNPasteImageFilter.h [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+ * need to copy the destination pixels to the output.
+ *
+ * When the source image is lower dimension than the destination image than the DestinationSkipAxes parameter specifies
+ * which axes in the destination image are set to 1 when copying the region o
```
Comment (zivy) [inline include/itkNPasteImageFilter.h]: Not sure what is the small 'o' after region?

File: include/itkNPasteImageFilter.h [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+  itkSetMacro(DestinationIndex, InputImageIndexType);
+  itkGetConstMacro(DestinationIndex, InputImageIndexType);
+
+  /** Set/Get the an array describing which axes in the destination image to skip
+   *
+   * The axes with true values are set to 1, to fill the difference between the dimension of the input and source image.
+   * The number of true value in DestinationSkipAxes plus the DestinationImageDimension must equal the
+   * InputImageDimension.
+   *
+   * By default this array contains SourceImageDimension false values followed filled with values for the remainder.
```
Comment (zivy) [inline include/itkNPasteImageFilter.h]: By default this array contains SourceImageDimension false values followed filled with values for the remainder. ->
By default this array contains SourceImageDimension false values followed by true values for the remainder.

File: include/itkNPasteImageFilter.h [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+  itkGetConstMacro(DestinationSkipAxes, InputSkipAxesArrayType);
+
+
+  /** Set/Get the source region (what part of the second input will be
+   * pasted. */
+  itkSetMacro(SourceRegion, SourceImageRegionType);
+  itkGetConstMacro(SourceRegion, SourceImageRegionType);
+
+  /** Set/Get the "destination" image.  This is the image that will be
+   * obscured by the paste operation. */
```
Comment (zivy) [inline include/itkNPasteImageFilter.h:137]: obscured -> overwritten or pasted

**Review by zivy — APPROVED** [review]
Really liked the image==hash_value testing, elegant.

File: include/itkNPasteImageFilter.h (hunk 4 above)
Comment (blowekamp) [reply include/itkNPasteImageFilter.h:137]: This is the original documentation.

**Pull request merged**